codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java

```diff
@@ -147,8 +147,7 @@ protected String getBaseDirectory() {
 
     @Override
     protected String getDiskFilename() {
-        File file = new File(filename);
-        return file.getName();
+        return "upload";
```

Comment:
There won't be any collisions.

The only usage of the changed method getDiskFilename() is in AbstractDiskHttpData.java tempFile(). There it is handed over to File.createTempFile as part of the file name. File.createTempFile will make sure there is no collision.

My first PR added an arbitrary counter to reduce the work File.createTempFile has to do, but @normanmaurer preferred the static string.